src/InterfaceLimits.jl

```diff
@@ -483,56 +573,61 @@ function find_interface_limits(
     return df
 end
 
-function find_monolithic_interface_limits(
+function find_interface_limits(
     sys::System,
-    solver;
-    branch_filter = x -> get_available(x),
-    ptdf = VirtualPTDF(sys),
-    lodf = nothing,
-    c_branches = nothing,
-    security = false,
-    enforce_load_distribution = false,
-    enforce_gen_limits = false,
+    solver::JuMP.MOI.OptimizerWithAttributes,
+    interface_key::Pair{String,String},
+    interface::Vector{ACBranch};
+    branch_filter::Function = x -> get_available(x),
+    ptdf::VirtualPTDF = VirtualPTDF(sys),
+    security::Union{Bool,Security} = false,
+    enforce_gen_limits::Bool = false,
+    enforce_load_distribution::Bool = false,
+    hops::Int = 3,
 )
-    # Build a JuMP Model
-    @info "Building interface limit optimization model"
-    m = direct_model(solver)
+    # Line hops:
+    in_branches, bus_neighbors =
+        find_neighbor_lines(sys, interface_key, branch_filter, hops)
 
-    interfaces = find_interfaces(sys, branch_filter)
-    in_branches = get_components(branch_filter, ACBranch, sys) # could filter for monitored lines here
-    gen_buses = Set(get_bus.(get_components(get_available, Generator, sys)))
-    load_buses = Set(get_bus.(get_components(get_available, LOAD_TYPES, sys)))
+    if security == true
+        security = Security(sys, get_components(x->(x ∈ in_branches),ACBranch,sys))
+    end
```

Comment:
I updated the default security to only consider in_branches instead of all ACBranches in the system since contingency_branches gets filtered down to the intersection between contingency_branches and in_branches later anyway. Should we keep the default like this or revert to original security=Security(sys) which includes all branches in the security struct?

Reply:
I think this is fine. I was going to suggest that we change the default `Security` constructor, but that would require a lot more changes since it doesn't have access to `in_branches`.